libnd4j/include/ops/declarable/platform/armcompute/armcomputeUtils.cpp

```diff
@@ -0,0 +1,291 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Konduit K.K.
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ******************************************************************************/
+
+ // Created by Abdelrauf 2020
+
+
+#include <ops/declarable/PlatformHelper.h>
+#include <ops/declarable/OpRegistrator.h>
+#include <system/platform_boilerplate.h> 
+#include <ops/declarable/helpers/convolutions.h>
+#include <cstdint>
+#include <helpers/LoopsCoordsHelper.h>
+
+#include "armcomputeUtils.h"
+
+
+namespace sd      {
+namespace ops       {
+namespace platforms {
+
+
+
+Arm_DataType getArmType ( const DataType &dType){
+     Arm_DataType  ret;
+     switch (dType){  
+        case HALF :
+            ret = Arm_DataType::F16;
+            break;        
+        case FLOAT32 :
+            ret = Arm_DataType::F32;
+            break;
+        case DOUBLE :
+            ret = Arm_DataType::F64;
+            break;
+        case INT8 :
+            ret = Arm_DataType::S8;
+            break;
+        case INT16 :
+            ret = Arm_DataType::S16;
+            break;
+        case INT32 :
+            ret = Arm_DataType::S32;
+            break;
+        case INT64 :
+            ret = Arm_DataType::S64;
+            break;
+        case UINT8 :
+            ret = Arm_DataType::U8;
+            break;
+        case UINT16 :
+            ret = Arm_DataType::U16;
+            break;        
+        case UINT32 :
+            ret = Arm_DataType::U32;
+            break;        
+        case UINT64 :
+            ret = Arm_DataType::U64;
+            break; 
+        case BFLOAT16 : 
+            ret = Arm_DataType::BFLOAT16;
+            break;
+        default:
+            ret = Arm_DataType::UNKNOWN;
+     }; 
+
+    return ret;
+}
+bool isArmcomputeFriendly(const NDArray& arr) {
+  auto dType = getArmType(arr.dataType());
+  int rank = (int)(arr.rankOf());
+  auto arrStrides = arr.stridesOf();
+  return dType != Arm_DataType::UNKNOWN && arr.ordering() == 'c' &&
+         arrStrides[rank - 1] == 1;
```

Comment:
> arrStrides[rank - 1] == 1;

As far as I understand you want to make sure array isn't permuted. That's wrong check in this case.
You should check for ews being equal to 1 AND shape::strideDescendingCAscendingF() == true

Reply:
C order and last rank strides being 1 were for the purpose of importing memory possible. but now as we have copy options of ndArray  we can make it redundant and add subarray copy for that case too.

Reply:
Sure, I'm just saying that you must ensure array isn't permuted. In such case this check will be insufficient.

Reply:
for now this cases dismissed with that check: